src/Keboola/DbExtractor/Retry/DbRetryPolicy.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractor;
+
+use Keboola\DbExtractor\Retry\RetryableException;
+use Retry\Policy\CallableRetryPolicy;
+
+class DbRetryPolicy extends CallableRetryPolicy
+{
+
+    public const DEFAULT_RETRYABLE_EXCEPTIONS = [
+        \PDOException::class => ['^42.*'], // ignore all 42xxx level error codes
+        \ErrorException::class => [] // don't ignore any ErrorExceptions
+    ];
+
+    public const DEFAULT_MAX_ATTEMPTS = 5;
+
+    /** @var  array RetryableException[] */
+    private $retryableExcpetions;
+
+    public function __construct(?int $maxAttempts = null, ?array $retryableExceptions = null)
+    {
+        if (isset($retryableExceptions)) {
+            $this->retryableExcpetions = $retryableExceptions;
+        } else {
+            $this->retryableExcpetions = array_map(
+                function ($key, $value) {
+                    return new RetryableException($key, $value);
```

Comment:
i'm not sure whether it's worth wrapping it into a class of its own? the method `shouldThisExceptionBeRetried` could easily be part of the decider? 
or maybe it's the naming - that i expect that `RetryableException` is an actual exception and it's really not.
and maybe it's because if feel that the checking is split between decide (checks exception) and `shouldThisExceptionBeRetried` checks message/code ?